src/lib/support/TestPersistentStorageDelegate.h

```diff
@@ -45,12 +46,19 @@ class TestPersistentStorageDelegate : public PersistentStorageDelegate
         VerifyOrReturnError(contains, CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND);
 
         std::vector<uint8_t> & value = mStorage[key];
-        uint16_t value_size          = static_cast<uint16_t>(value.size());
-        VerifyOrReturnError(value_size <= size, CHIP_ERROR_BUFFER_TOO_SMALL);
-
-        size = std::min(value_size, size);
-        memcpy(buffer, value.data(), size);
-        return CHIP_NO_ERROR;
+        size_t valueSize             = value.size();
+        if (size < valueSize)
+        {
+            size = CanCastTo<uint16_t>(valueSize) ? static_cast<uint16_t>(valueSize) : 0;
+            return CHIP_ERROR_BUFFER_TOO_SMALL;
+        }
+        else
+        {
+            size = static_cast<uint16_t>(valueSize);
+            VerifyOrReturnError(buffer != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
This does not seem consistent with the Server.h delegate impl in terms of what happens if buffer is null and the incoming size is 0...  And the API documentation does not say what will happen in that case.

Reply:
Will fix the docs